logstash-core-event-java/src/main/java/com/logstash/ext/JrubyEventExtLibrary.java

```diff
@@ -251,9 +251,24 @@ public IRubyObject ruby_to_java(ThreadContext context)
 
         @JRubyMethod(name = "to_json", rest = true)
         public IRubyObject ruby_to_json(ThreadContext context, IRubyObject[] args)
-            throws IOException
         {
-            return RubyString.newString(context.runtime, event.toJson());
+            try {
+                return RubyString.newString(context.runtime, event.toJson());
+            } catch (Exception e) {
+                throw new RaiseException(context.runtime, context.runtime.getModule("LogStash").defineOrGetModuleUnder("Json").getClass("GeneratorError"), e.getMessage(), true);
```

Comment:
it is and AFAIK this is the only way to get a reference of an inner module. maybe this could be established earlier on in the Library load so that if this namespace does not exist it will crash on startup and not at exception time.  thanks, good observation.